### PR TITLE
OIDF - 5

### PR DESCRIPTION
**Pull request by robertmathew**
Changes:

Map a Entity Stement from JWT to OpenAPI model in KMP

Map a Trustchain from JWT to OpenAPI model in KMP

Assign the category (leaf, TA, intermediate) to the trustchain array items

### CODE REVIEW

File: modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/logic/EntityLogic.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        return entityStatement.metadata?.federationEntity?.federationListEndpoint?.isNotEmpty() == true
+    }
+}
+
+enum class EntityType {
+    LEAF,
+    INTERMEDIATE,
+    TRUST_ANCHOR,
+    UNDEFINED
+}
```
Comment (maikel-maas) [inline modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/logic/EntityLogic.kt:33]: Add trailing new line

File: modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/mapper/JsonMapper.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        return try {
+            Triple(
+                Json.decodeFromString(headerJson), Json.parseToJsonElement(payloadJson), JWTSignature(parts[2])
+            )
+        } catch (e: Exception) {
+            println(e.printStackTrace())
+            Triple(null, null, null)
+        }
+    }
+}
```
Comment (maikel-maas) [inline modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/mapper/JsonMapper.kt:59]: Add trailing new line

File: modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/model/JWTHeader.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+)
```
Comment (maikel-maas) [inline modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/model/JWTHeader.kt:13]: Add trailing new line

File: modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/model/JWTSignature.kt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package com.sphereon.oid.fed.common.model
+
+
+data class JWTSignature(val value: String)
```
Comment (maikel-maas) [inline modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/model/JWTSignature.kt:4]: Add trailing new line

File: modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/logic/EntityLogicTest.kt [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Test
+    fun shouldReturnUndefined() {
+        val entityStatement = EntityStatement(
+            metadata = Metadata(federationEntity = null),
+            authorityHints = emptyList()
+        )
+        assertEquals(EntityType.UNDEFINED, entityLogic.getEntityType(entityStatement))
+    }
+}
```
Comment (maikel-maas) [inline modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/logic/EntityLogicTest.kt:56]: Add trailing new line

File: modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/mapper/JsonMapperTest.kt [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    fun testDecodeJWTWithInvalidJSON() {
+        val jwtWithInvalidJson =
+            "eyJraWQiOiJCNkVCODQ4OENDODRDNDEwMTcxMzRCQzc3RjQxMzJBMDQ2N0NDQzBFIiwidHlwIjoiZW50aXR5LXN0YXRlbWVudFx1MDAyQmp3dCIsImFsZyI6IlJTMjU2In0.eyJzdWI6IjEyMzQ1Njc4OTAiLCJuYW1lIjoiSm9obiBEb2UiLCJhZG1pbiI6dHJ1ZX0.NHVaYe26MbtOYhSKkoKYdFVomg4i8ZJd8_-RU8VNbftc" // Missing quote in payload
+        val (header, payload, signature) = mapper.decodeJWTComponents(jwtWithInvalidJson)
+
+        assertNull(header)
+        assertNull(payload)
+        assertNull(signature)
+    }
+}
```
Comment (maikel-maas) [inline modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/mapper/JsonMapperTest.kt:52]: Add trailing new line

File: settings.gradle.kts [hunk 7]
```diff
@@ -24,10 +25,11 @@ dependencyResolutionManagement {
...
         }
+        mavenLocal()
         mavenCentral()
     }
 }
 
 include(":modules:openid-federation-common")
 include(":modules:admin-server")
-include(":modules:openapi")
+include(":modules:openapi")
```
Comment (maikel-maas) [inline settings.gradle.kts:35]: Add trailing new line

File: modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/logic/EntityLogicTest.kt [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
..."sig","n":"yF7zakG7N8G2meDjsgqFowtxE51FtG8HacCDxA_qiMGp5lrcC4CggEdHkpdI3RrYTZ9WvADMWZEdIwydbIPn1HsPNCEy1UpsWQEZ-_8mHXGVz6p7ePrPXjIKl2eGqyQG_iCdAjqVpQ43Fqe3Mg17V-Phnn3gN_zMS7-eOXwuhWPiLZn7mFkiGin2LKECok4aaOcxIWRhsiNJdT5j0T1ORuk9y-gfJ-ljauVLh4hn5cwz7nj2lKBA0e-FOBTDSbpL_jYCj0NkvCrgFJSPDitne4M_M6YM1GpSNuBNNCOGYzjmynGpqu3btgdd5ONh2Ym4Kzwspu_RnLY3lyWR6lC_ad7q4fl4Zjlhp6murcKt15OkvijFVILraFqglugP6lHb89j1QQKKeBRNj_k93YQ5o1cT3iV-j-2BMcgB5Rfp-1sKlz10QQlMWXnbN7ruKs6Q2M5XNOGu8fJ9ohYM9rECmPmmVQI5vzoH65JfbKT0Mgfer59QY1s4IEgn5csGEw2p","e":"AQAB","kid":"B6EB8488CC84C41017134BC77F4132A0467CCC0E"},{"kty":"RSA","use":"enc","n":"yF7zakG7N8G2meDjsgqFowtxE51FtG8HacCDxA_qiMGp5lrcC4CggEdHkpdI3RrYTZ9WvADMWZEdIwydbIPn1HsPNCEy1UpsWQEZ-_8mHXGVz6p7ePrPXjIKl2eGqyQG_iCdAjqVpQ43Fqe3Mg17V-Phnn3gN_zMS7-eOXwuhWPiLZn7mFkiGin2LKECok4aaOcxIWRhsiNJdT5j0T1ORuk9y-gfJ-ljauVLh4hn5cwz7nj2lKBA0e-FOBTDSbpL_jYCj0NkvCrgFJSPDitne4M_M6YM1GpSNuBNNCOGYzjmynGpqu3btgdd5ONh2Ym4Kzwspu_RnLY3lyWR6lC_ad7q4fl4Zjlhp6murcKt15OkvijFVILraFqglugP6lHb89j1QQKKeBRNj_k93YQ5o1cT3iV-j-2BMcgB5Rfp-1sKlz10QQlMWXnbN7ruKs6Q2M5XNOGu8fJ9ohYM9rECmPmmVQI5vzoH65JfbKT0Mgfer59QY1s4IEgn5csGEw2p","e":"AQAB","kid":"B6EB8488CC84C41017134BC77F4132A0467CCC0E_enc"}]}}"""
```
Comment (maikel-maas) [inline modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/logic/EntityLogicTest.kt:13]: Add space between the constants

Comment (maikel-maas) [inline modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/logic/EntityLogicTest.kt:13]: Add space between the constants

File: modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/logic/EntityLogic.kt (hunk 1 above)
Comment (robertmathew) [reply modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/logic/EntityLogic.kt:33]: Are we following any specific style? Isn't this style for Python?

File: modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/model/JWTHeader.kt [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package com.sphereon.oid.fed.common.model
+
+import kotlinx.serialization.Serializable
+
+import kotlinx.serialization.SerialName
+
+@Serializable
+data class JWTHeader(
```
Comment (maikel-maas) [inline modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/model/JWTHeader.kt:8]: Can we move it to the Open Api models?

**Review by maikel-maas — CHANGES_REQUESTED** [review]
Left some remarks

File: modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/logic/EntityLogic.kt (hunk 1 above)
Comment (robertmathew) [reply modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/logic/EntityLogic.kt]: I have made the changes

**Comment by robertmathew** [comment]
I have made the necessary changes.

File: modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/model/JWTHeader.kt (hunk 9 above)
Comment (robertmathew) [reply modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/model/JWTHeader.kt]: Yes. I have made the changes

File: modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/logic/EntityLogicTest.kt (hunk 8 above)
Comment (robertmathew) [reply modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/logic/EntityLogicTest.kt]: I have created a constant file and moved it.

File: modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/mapper/JsonMapperTest.kt (hunk 6 above)
Comment (robertmathew) [reply modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/mapper/JsonMapperTest.kt]: I have made the changes

File: modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/model/JWTSignature.kt (hunk 4 above)
Comment (robertmathew) [reply modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/model/JWTSignature.kt]: I have moved it to OpenAPI models

File: modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/logic/EntityLogic.kt (hunk 1 above)
Comment (maikel-maas) [reply modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/logic/EntityLogic.kt]: Yes, Google Style. AFAIK it's not Python specific

**Review by sanderPostma — APPROVED** [review]

File: modules/openid-federation-common/build.gradle.kts [hunk 10]
```diff
@@ -40,16 +40,21 @@ kotlin {
         }
     }
 
-    iosX64()
-    iosArm64()
-    iosSimulatorArm64()
+//    iosX64()
```
[the remark this answers is not in the document]
Comment (sanderPostma) [reply modules/openid-federation-common/build.gradle.kts:43]: LGTM
The  iosX64() is being resolved by Zoe I see

**Review by maikel-maas — APPROVED** [review]
LGTM

**Pull request merged**